### PR TITLE
Fixes #6424: Correct and harmonize check-rsyslog-version and check-zyppe...

### PR DESCRIPTION
**Pull request by Kegeruneku**
...r-version

Ticket: http://www.rudder-project.org/redmine/issues/6424

To be merged in 2.10

### CODE REVIEW

File: tools/check-rsyslog-version [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+RPMVERCMP="/var/rudder/cfengine-community/bin/rpmvercmp"
+
+# Minimum supported Rsyslog version
+MINIMAL_RSYSLOG_VERSION="5.7.1"
+
+if type rsyslogd >/dev/null 2>&1; then
+
+  CURRENT_RSYSLOG_VERSION=$(rsyslogd -v | head -n1 | sed "s/^rsyslogd \\([^, ]*\\).*$/\\1/")
+
+  if ${RPMVERCMP} ${MINIMAL_RSYSLOG_VERSION} lt ${CURRENT_RSYSLOG_VERSION}; then
```
Comment (ncharles) [inline tools/check-rsyslog-version]: could you check the semantic here of lt ?

File: tools/check-zypper-version [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RPMVERCMP="/var/rudder/cfengine-community/bin/rpmvercmp"
+
+# Minimum supported Zypper version
+MINIMAL_ZYPPER_VERSION="1.0.8"
+
+if type zypper >/dev/null 2>&1; then
+
+  CURRENT_ZYPPER_VERSION=$(/usr/bin/zypper --version | cut -d' ' -f2)
+
+  if ${RPMVERCMP} ${MINIMAL_ZYPPER_VERSION} lt ${CURRENT_ZYPPER_VERSION}; then
```
Comment (jooooooon) [inline tools/check-zypper-version]: What happens if they are equal? Don't you want "le" rather than "lt"?

File: tools/check-zypper-version [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    echo "+zypper_version_ok"
+  else
+    echo "+zypper_version_not_ok"
```
Comment (ncharles) [inline tools/check-zypper-version]: you changed the name of the class ? it is however used in several techniques

Comment (ncharles) [inline tools/check-zypper-version]: we should not change on a stable version the classes define, people may have use it in their own technique

Comment (Kegeruneku) [inline tools/check-zypper-version]: Well, let's output both classes then :)

File: tools/check-zypper-version (hunk 2 above)
Comment (Kegeruneku) [inline tools/check-zypper-version]: There is no le, I'll have to add another condition for equality... :)

Comment (Kegeruneku) [inline tools/check-zypper-version]: Or as @ncharles suggested, just reverse the condition :)

**Comment by jooooooon** [comment]
Aside from the inline comments, this is an AWESOME improvement. I love seeing cleanups like this that make code simpler, and renaming files to match our convention.

**Comment by Kegeruneku** [comment]
Thanks :D

**Comment by Kegeruneku** [comment]
PR updated according to review !

**Comment by peckpeck** [comment]
I don't see in your PR that you reverted the condition, nor that you included the old classes in the output. 
Is this the last version ?

**Comment by Kegeruneku** [comment]
Yup, it is :)

**Comment by Kegeruneku** [comment]
Old one was:
```
if ${RPMVERCMP} ${MINIMAL_ZYPPER_VERSION} lt ${CURRENT_ZYPPER_VERSION}; then
```

new one is:
```
if ${RPMVERCMP} ${CURRENT_ZYPPER_VERSION} lt ${MINIMAL_ZYPPER_VERSION}; then
```

**Comment by Kegeruneku** [comment]
Also, I added:
```
# Also output the legacy variable name, just in case
echo "+zypper_version_unusable"
```

The other tool outputs "+rsyslog_greater_than_5_7_1" which is way too specific and IMHO, there is no need to keep the legacy ones is it is purely internal (promises.st)

**Comment by peckpeck** [comment]
That's OK for me

**Pull request merged**